src/lib/PDFViewer/PageInternals/PageCanvas.svelte

```diff
@@ -23,7 +28,9 @@
 
 		try {
 			await render_task.promise;
+			dispatch('pageloadsuccess', page);
 		} catch (err) {
+			dispatch('pageloaderror', err);
 			if (!(err instanceof RenderingCancelledException)) throw err;
 		}
```

Comment:
```suggestion
			if (!(err instanceof RenderingCancelledException)) {
				dispatch('pageloaderror', err);
				throw err;
			}
		}
```

Can you move the dispatch for the error event inside that if statement?
`RenderingCancelledException` is more of an expected exception when the page's rendering is cancelled by changing the page in the middle of rendering.

Also, I think it'd be better to name these as `pagerendersuccess` and `pagerendererror` instead since they're both emitted depending on the success/failure of rendering the page to the canvas.